### PR TITLE
benchmark: add boon

### LINKED ISSUE #5: benchmark: add boon
https://github.com/santhosh-tekuri/boon

### PR DESCRIPTION
**Pull request by tony-go**
Fix #5

### CODE REVIEW

File: implementations/boon/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  let file = File::open(&instance_file)?;
+  let reader = BufReader::new(file);
+
+  // Compile the schema
+  let mut schemas = Schemas::new();
+  let mut compiler = Compiler::new();
+  let sch_index = compiler.compile(schema_file.to_str().ok_or("NULL")?, &mut schemas)?;
+
+  // Validate each instance
+  let start = Instant::now();
```
Comment (jviotti) [inline implementations/boon/src/main.rs:25]: One improvement you should do there, is that you are doing I/O on the JSONL file, including parsing each line with Serde, on the code block that is measuring the benchmark, which can affect results.

You likely should:

- First parse the JSONL and each of its entries
- Store them in an array/vector or whatever Rust uses
- THEN start measuring, loop on the parsed list, and validate

File: implementations/boon/Cargo.toml
Comment (jviotti) [inline implementations/boon/Cargo.toml:1]: Once you rebase, you will be expected to create `implementations/boon/version.sh` that the only thing it does is print the version. You can implement it with some `sed`/`awk` on this file to get the `0.6`

File: Makefile [hunk 2]
```diff
@@ -40,3 +40,14 @@ dist/results/ajv/%: \
...
+
+# BOON
+
+dist/results/boon/%: \
+	implementations/boon/src/main.rs \
+	implementations/boon/Cargo.toml \
+	schemas/%/schema.json \
+	schemas/%/instances.jsonl \
+	| dist/results/boon
+	cd implementations/boon && cargo run --release ../../$(dir $(word 3,$^)) > ../../$@
```
Comment (jviotti) [inline Makefile:52]: I think the `cd` here is making things more complicated than they should, with `../../`, etc. From a quick look at the man, looks like you can do `--manifest-path implementations/boon/Cargo.toml`

File: implementations/boon/src/main.rs (hunk 1 above)
Comment (tony-go) [reply implementations/boon/src/main.rs:25]: Damn! I let that! Thanks ❤️

File: Makefile (hunk 2 above)
Comment (tony-go) [reply Makefile:52]: yeah! It did not bother me on first place but its much better!

File: .DS_Store
Comment (jviotti) [inline .DS_Store:1]: Can we add this to `.gitignore`?

File: .gitignore [hunk 3]
```diff
@@ -1,2 +1,3 @@
 /dist
 /node_modules
+implementations/boon/target/
```
Comment (jviotti) [inline .gitignore:3]:
```suggestion
/implementations/boon/target/
```

Just to be explicit?

File: .DS_Store
Comment (tony-go) [reply .DS_Store:1]: Damn I forgot it! sorry!

**Review by jviotti — APPROVED** [review]

**Pull request merged**